### PR TITLE
add TensorView::DeviceMesh*

### PR DESCRIPTION
**Pull request by samnordmann**
Add a `DeviceMesh*` (private) attribute to the TensorView. This is part of the multidevice scheduling API. We add basic methods:
- setDeviceMesh
- getDeviceMesh
- hadDeviceMesh
and update the printing utility.

### CODE REVIEW

File: csrc/ir/interface_nodes.h [hunk 1]
```diff
@@ -600,6 +614,8 @@ class TensorView : public Val {
   //! current tensor. This will then allow us to safely reuse the memory
   //! allocated to this tensor.
   bool promote_reuse_ = false;
+
+  DeviceMesh* mesh_ = nullptr;
```
[the remark this answers is not in the document]
Comment (samnordmann) [reply csrc/ir/interface_nodes.h:618]: The idea is to make (in a near future) DeviceMesh a Val IR registered in the Fusion. So the Fusion will eventually own and manage lifetime, while TensorView should only holds a pointer. But I agree that right now this is incomplete. If you want I can add to this/seperate PR a patch that makes DeviceMesh an IR. what do you suggest ?

[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/ir/interface_nodes.h:618]: I'd keep it as simple as possible for this PR. Why not start with just a non-pointer `DeviceMesh` and then updating it as necessary?

[the remark this answers is not in the document]
Comment (samnordmann) [reply csrc/ir/interface_nodes.h]: ok, I changed as you suggest

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**